react/my-app/src/App.tsx

```diff
@@ -1,11 +1,20 @@
 import React from 'react';
 import logo from './logo.svg';
+import { useState } from 'react';
 import './App.css';
 import { Wrapper } from './hw_37/wrapper';
+import { ChangeTheme, ThemeContext } from './hw_41/themeContext';
 
 function App() {
+
+  const [theme, setTheme] = useState("light")
+
   return (
-    <Wrapper></Wrapper>
+    <ThemeContext.Provider value={theme}>
+      <ChangeTheme.Provider value={() => theme === "light" ? setTheme("dark") : setTheme("light")}>
```

Comment:
prefer to put Providers and state to another component
it will protect Wrapper from rerender
You can check it in ReactDevTools